src/ws/Socket.ts

```diff
@@ -593,7 +595,10 @@ export class Socket extends EventEmitter {
      */
     public optOutEvents(args: string[]): Promise<void> {
         this._optOutEventsArgs = args;
```

Comment:
this seems like an initialization config. Is there any reason why we are not passing this through the config in the constructor?

Reply:
I would like this method to be isolated and independent as possible and hence wouldn't bind this to `Socket` constructor.